### PR TITLE
Fix client file endpoint

### PR DESCRIPTION
**Pull request by mangelajo**
Fixes-Issue: #121

### CODE REVIEW

File: cmd/flightctl/main.go [hunk 1]
```diff
@@ -554,23 +558,31 @@ func applyFromReader(client *apiclient.ClientWithResponses, filename string, r i
...
 				continue
 			}
 			fmt.Printf("%s\n", response.HTTPResponse.Status)
+			httpResponse = *response.HTTPResponse
 		default:
 			errs = append(errs, fmt.Errorf("%s: skipping resource of unkown kind %q: %v", filename, kind, resource))
 		}
+
+		// bad HTTP Responses don't generate an error on the OpenAPI client, we need to check the status code manually
+		if httpResponse.StatusCode != http.StatusOK && httpResponse.StatusCode != http.StatusCreated {
```
Comment (hexfusion) [inline cmd/flightctl/main.go:583]: if we hit the default case would this panic?

File: cmd/flightctl/main.go [hunk 2]
```diff
@@ -554,23 +558,31 @@ func applyFromReader(client *apiclient.ClientWithResponses, filename string, r i
...
+			errs = append(errs, fmt.Errorf("%s: failed to apply device/%s: %v", filename, resourceName, httpResponse.Status))
```
Comment (fzdarsky) [inline cmd/flightctl/main.go:584]:
```suggestion
			errs = append(errs, fmt.Errorf("%s: failed to apply %s/%s: %v", filename, strings.ToLower(kind), resourceName, httpResponse.Status))
```

File: cmd/flightctl/main.go (hunk 1 above)
Comment (hexfusion) [reply cmd/flightctl/main.go:583]: no it will just be status 0 I thought it was a ref my bad :)

File: cmd/flightctl/main.go (hunk 2 above)
Comment (fzdarsky) [reply cmd/flightctl/main.go:584]: Maybe fixing lines 524 and 527 in the process :)

File: cmd/flightctl/main.go (hunk 1 above)
Comment (fzdarsky) [inline cmd/flightctl/main.go:583]: In the default case of the switch statement, httpResponse.StatusCode would be == 0, so produce two errors.

Comment (mangelajo) [reply cmd/flightctl/main.go]: ah but it would emit an additional error in such case. I'll fix that

Comment (mangelajo) [reply cmd/flightctl/main.go]: each call seems to return a different type, but I will try to factor out as much as I can.

Comment (mangelajo) [reply cmd/flightctl/main.go]: done

File: cmd/flightctl/main.go (hunk 2 above)
Comment (mangelajo) [reply cmd/flightctl/main.go]: done.

File: cmd/flightctl/main.go (hunk 1 above)
Comment (mangelajo) [reply cmd/flightctl/main.go]: done.

**Pull request merged**